src/NUnitFramework/framework/Internal/Execution/CommandBuilder.cs

```diff
@@ -52,19 +52,27 @@ public static TestCommand MakeOneTimeSetUpCommand(TestSuite suite, List<SetUpTea
             TestCommand command = new OneTimeSetUpCommand(suite, setUpTearDown, actions);
 
             // Prefix with any IApplyToContext items from attributes
+            IList<IApplyToContext> changes = null;
+
             if (suite.TypeInfo != null)
+                changes = suite.TypeInfo.GetCustomAttributes<IApplyToContext>(true);
+            else if (suite.Method != null)
+                changes = suite.Method.GetCustomAttributes<IApplyToContext>(true);
+            else
             {
-                IApplyToContext[] changes = suite.TypeInfo.GetCustomAttributes<IApplyToContext>(true);
-                if (changes.Length > 0)
-                    command = new ApplyChangesToContextCommand(command, changes);
-            }
-            if (suite.Method!=null)
-            {
-                IApplyToContext[] changes = suite.Method.GetCustomAttributes<IApplyToContext>(true);
-                if (changes.Length > 0)
-                    command = new ApplyChangesToContextCommand(command, changes);
+                var testAssembly = suite as TestAssembly;
+                if (testAssembly != null)
+#if PORTABLE
+                    // TODO: Fix this code! It's really quite ugly!
+                    changes = new List<IApplyToContext>(testAssembly.Assembly.GetAttributes<IApplyToContext>());
```

Comment:
This .NET 2.0. :-) Actually, the TODO was for my earlier code with a double cast.